### PR TITLE
Embedding chahub

### PR DESCRIPTION
**Pull request by ckcollab**
# TODO

- [ ] set models to "unpublish" or "hide" when appropriate #2266

### CODE REVIEW

File: codalab/apps/web/static/less/main.less [hunk 1]
```diff
@@ -328,3 +328,18 @@ starting_kit_td {
...
+    left: 0;
+    right: 0;
+    bottom: 0;
+    top: 0;
+    border: 0;
+
+    // Doing padding instead of top: 50px makes body not have a scrollbar
+    padding-top: 50px;
+    padding-bottom: 25px;
+}
```
Comment (Tthomas63) [inline codalab/apps/web/static/less/main.less]: No newline

File: codalab/apps/web/tasks.py [hunk 2]
```diff
@@ -834,6 +834,7 @@ def do_phase_migrations():
     logger.info("Checking {} competitions for phase migrations.".format(len(competitions)))
     for c in competitions:
         c.check_future_phase_sumbmissions()
+    logger.info("@@@ Done checking phase migrations!")
```
Comment (Tthomas63) [inline codalab/apps/web/tasks.py]: Do we need the `@@@` in this log statement?

File: codalab/codalab/context_processors.py [hunk 3]
```diff
@@ -21,4 +20,8 @@ def common_settings(request):
         'local_ace_editor': codalab_settings.LOCAL_ACE_EDITOR,
         'is_dev': codalab_settings.IS_DEV,
         'USE_AWS': codalab_settings.USE_AWS,
+
+        # Just get the base of the URL
+        'CHAHUB_URL': '//{}'.format(settings.CHAHUB_API_URL.split("//")[-1].split("/")[0].split('?')[0]),
```
Comment (Tthomas63) [inline codalab/codalab/context_processors.py]: Should we have a function or something nicer format this?

File: codalab/apps/chahub/models.py [hunk 4]
```diff
@@ -118,4 +118,4 @@ def save(self, force_to_chahub=False, *args, **kwargs):
                     # We save at the beginning, but then again at the end to save our new chahub timestamp and such
                     super(ChaHubSaveMixin, self).save(*args, **kwargs)
             else:
-                logger.info("ChaHub :: Model failed validation")
+                pass
```
Comment (Tthomas63) [inline codalab/apps/chahub/models.py]: Should we still have a log indicating it did not pass validation?

**Comment by Tthomas63** [comment]
Tests pass locally? (CircleCI is down at the moment)

File: codalab/codalab/context_processors.py (hunk 3 above)
Comment (ckcollab) [reply codalab/codalab/context_processors.py]: yes and that function should have tests! :)

File: codalab/apps/web/tasks.py (hunk 2 above)
Comment (ckcollab) [reply codalab/apps/web/tasks.py]: We don't need it, for sure, but definitely helps out when skimming the log

**Pull request closed without merging**